### PR TITLE
fix(ci): update mysql.connector version

### LINKED ISSUE #8038: test: test_sqllogic_standalone_macos datetime test always failed
**Summary**

```
Following failed statements:
---------------------------------------------
Runner: mysql
ErrorType: statement query get result not equal to expected
Message: 
 Expected:
2021-09-05 09:23:17.000000 1                                                    
 Actual:
2021-09-05 09:23:17  1                                                          
 Statement:
Parsed Statement
    at_line: 23,
    s_type: Statement: query, type: TB, query_type: TB, retry: False,
    suite_name: base/02_function/02_0012_function_datetimes,
    text:
        select to_datetime(1630833797), to_int64(to_datetime(1630833797)) = 1630833797000000;
    results: [(<re.Match object; span=(0, 4), match='----'>, 27, '2021-09-05 09:23:17.000000 1')],
    runs_on: {'http', 'mysql', 'clickhouse'},
 Start Line: 27, Result Label: 
---------------------------------------------
Runner: mysql
ErrorType: statement query get result not equal to expected
Message: 
 Expected:
2021-08-30 10:47:42.000000                                                      
 Actual:
2021-08-30 10:47:42                                                             
 Statement:
Parsed Statement
    at_line: 7,
    s_type: Statement: query, type: T, query_type: T, retry: False,
    suite_name: base/02_function/02_0012_function_datetimes_tz,
    text:
        select to_timestamp(1630320462000000);
    results: [(<re.Match object; span=(0, 4), match='----'>, 11, '2021-08-30 10:47:42.000000')],
    runs_on: {'http', 'mysql', 'clickhouse'},
 Start Line: 11, Result Label: 
---------------------------------------------
Runner: mysql
ErrorType: statement query get result not equal to expected
Message: 
 Expected:
2022-04-23 2022-04-23 01:48:19.000000                                           
 Actual:
2022-04-23  2022-04-23 01:48:19                                                 
 Statement:
Parsed Statement
    at_line: 201,
    s_type: Statement: query, type: TT, query_type: TT, retry: False,
    suite_name: base/02_function/02_0045_conditional_function_in,
    text:
        SELECT * FROM t2 WHERE date in ('2022-04-23');
    results: [(<re.Match object; span=(0, 4), match='----'>, 205, '2022-04-23 2022-04-23 01:48:19.000000')],
    runs_on: {'http', 'mysql', 'clickhouse'},
 Start Line: 205, Result Label: 
---------------------------------------------
Runner: mysql
ErrorType: statement query get result not equal to expected
Message: 
 Expected:
2022-04-01 00:00:00.000000 5
2022-04-01 00:00:00.000001 5                       
 Actual:
2022-04-01 00:00:00  5
2022-04-01 00:00:00.000001  5                            
 Statement:
Parsed Statement
    at_line: 97,
    s_type: Statement: query, type: TI, query_type: TI, retry: False,
    suite_name: base/03_dml/03_0003_select_group_by,
    text:
        select created_time, sum(count) from t_datetime group by created_time order by created_time;
    results: [(<re.Match object; span=(0, 4), match='----'>, 102, '2022-04-01 00:00:00.000000 5\n2022-04-01 00:00:00.000001 5')],
    runs_on: {'http', 'mysql', 'clickhouse'},
 Start Line: 102, Result Label: 
---------------------------------------------
Runner: mysql
ErrorType: statement query get result not equal to expected
Message: 
 Expected:
1 2 3 4 2021-08-15 2021-09-15 2021-08-15 10:00:00.000000 string1234
5 6 7 8 2021-10-15 2021-11-15 2021-11-15 10:00:00.000000 string5678
 Actual:
1  2  3  4  2021-08-15  2021-09-15  2021-08-15 10:00:00  string1234
5  6  7  8  2021-10-15  2021-11-15  2021-11-15 10:00:00  string5678
 Statement:
Parsed Statement
    at_line: 31,
    s_type: Statement: query, type: IIIITTTT, query_type: IIIITTTT, retry: False,
    suite_name: base/03_dml/03_0014_insert_into_select,
    text:
        SELECT * FROM t1;
    results: [(<re.Match object; span=(0, 4), match='----'>, 36, '1 2 3 4 2021-08-15 2021-09-15 2021-08-15 10:00:00.000000 string1234\n5 6 7 8 2021-10-15 2021-11-15 2021-11-15 10:00:00.000000 string5678')],
    runs_on: {'http', 'mysql', 'clickhouse'},
 Start Line: 36, Result Label: 
---------------------------------------------
Runner: mysql
ErrorType: statement query get result not equal to expected
Message: 
 Expected:
-1 33 2021-08-15 2021-08-15 10:00:00.000000 string1234 140000.0
101 67 2021-11-15 2021-11-15 10:00:00.000000 string5678 0.0099
101 67 2021-11-15 2021-11-15 10:00:00.000000 string5678 9900.0
100 100 1970-01-01 1970-01-01 00:00:00.000000 string7890 12.34
100 100 1000-01-01 1000-01-01 00:00:00.000000 stringabcd 56.78
 Actual:
-1  33  2021-08-15  2021-08-15 10:00:00  string1234  140000.0
101  67  2021-11-15  2021-11-15 10:00:00  string5678  0.0099
101  67  2021-11-15  2021-11-15 10:00:00  string5678  9900.0
100  100  1970-01-01  1970-01-01 00:00:00  string7890  12.34
100  100  1000-01-01  1000-01-01 00:00:00  stringabcd  56.78
 Statement:
Parsed Statement
    at_line: 16,
    s_type: Statement: query, type: IITTTF, query_type: IITTTF, retry: False,
    suite_name: base/03_dml/03_0016_insert_into_values,
    text:
        select * from t1;
    results: [(<re.Match object; span=(0, 4), match='----'>, 24, '-1 33 2021-08-15 2021-08-15 10:00:00.000000 string1234 140000.0\n101 67 2021-11-15 2021-11-15 10:00:00.000000 string5678 0.0099\n101 67 2021-11-15 2021-11-15 10:00:00.000000 string5678 9900.0\n100 100 1970-01-01 1970-01-01 00:00:00.000000 string7890 12.34\n100 100 1000-01-01 1000-01-01 00:00:00.000000 stringabcd 56.78')],
    runs_on: {'http', 'mysql', 'clickhouse'},
 Start Line: 24, Result Label: 
---------------------------------------------
Runner: mysql
ErrorType: statement query get result not equal to expected
Message: 
 Expected:
2021-01-01 01:01:01.000000 2022-01-01 01:01:01.000000
1990-12-01 10:11:12.000000 2030-01-12 22:00:00.000000
 Actual:
2021-01-01 01:01:01  2022-01-01 01:01:01
1990-12-01 10:11:12  2030-01-12 22:00:00
 Statement:
Parsed Statement
    at_line: 280,
    s_type: Statement: query, type: TT, query_type: TT, retry: False,
    suite_name: base/03_dml/03_0023_insert_into_array,
    text:
        select arr[0], arr[1] from t13;
    results: [(<re.Match object; span=(0, 4), match='----'>, 285, '2021-01-01 01:01:01.000000 2022-01-01 01:01:01.000000\n1990-12-01 10:11:12.000000 2030-01-12 22:00:00.000000')],
    runs_on: {'http', 'mysql', 'clickhouse'},
 Start Line: 285, Result Label: 
---------------------------------------------
```

https://github.com/datafuselabs/databend/actions/runs/3195826596/jobs/5234949892#step:3:2064

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

update mysql.connector version

Fixes #8038

### CODE REVIEW

File: tests/logictest/requirements.txt [hunk 1]
```diff
@@ -1,9 +1,9 @@
-# TODO: specify version of dependencies
-mysql-connector
-six
-PyHamcrest
-requests
-environs
+# This file need to be manually keeped update with pyproject.toml
```
Comment (TCeason) [inline tests/logictest/requirements.txt:1]: In the ci logical test the docker use this docker file: tests/logictest/Dockerfile https://github.com/datafuselabs/databend/blob/main/tests/logictest/Dockerfile

If only modifying the requirements.txt, It maybe not be valid.

Comment (sundy-li) [reply tests/logictest/requirements.txt:1]: ok, the dockerfile should use `requirements.txt` instead.

**Review by TCeason — APPROVED** [review]

**Comment by sundy-li** [comment]
#8040 8040

**Comment by sundy-li** [comment]
@mergify update

**Review by BohuTANG — APPROVED** [review]

**Comment by BohuTANG** [comment]
Docker image build failure for the main branch:
https://github.com/datafuselabs/databend/actions/runs/3209192379/jobs/5245703818#step:4:199

```
#8 [3/7] RUN mkdir /test
#8 0.144 mkdir: cannot create directory ‘/test’: File exists
#8 ERROR: process "/bin/sh -c mkdir /test" did not complete successfully: exit code: 1
------
 > [3/7] RUN mkdir /test:
#8 0.144 mkdir: cannot create directory ‘/test’: File exists
------
Dockerfile:5
--------------------
   3 |     WORKDIR /test
   4 |     
   5 | >>> RUN mkdir /test
   6 |     COPY *.py /test
   7 |     COPY requirements.txt /test
--------------------
ERROR: failed to solve: process "/bin/sh -c mkdir /test" did not complete successfully: exit code: 1
Error: buildx failed with: ERROR: failed to solve: process "/bin/sh -c mkdir /test" did not complete successfully: exit code: 1

```

**Comment by BohuTANG** [comment]
It seems not to work for the main branch:
https://github.com/datafuselabs/databend/actions/runs/3211012023/jobs/5249318501#step:3:2073

cc @sundy-li

**Pull request merged**